### PR TITLE
Week 1

### PR DESCRIPTION
**Pull request by Erkan-5Minds**

### CODE REVIEW

File: Week_1_C/cash/cash.c [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#include <stdio.h>
+#include <cs50.h>
+#include <math.h>
+
```
Comment (DevBastiS) [inline Week_1_C/cash/cash.c:4]:
```suggestion
```

File: Week_1_C/cash/cash.c [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+int main(void)
+{
+
```
Comment (DevBastiS) [inline Week_1_C/cash/cash.c]:
```suggestion
```

File: Week_1_C/credit/credit.c [hunk 3]
```diff
@@ -0,0 +1,78 @@
+#include <cs50.h>
+#include <stdio.h>
+#include <math.h>
+
+long long cardNumber = 0;
+
+int main(void)
+{
+
```
Comment (DevBastiS) [inline Week_1_C/credit/credit.c]:
```suggestion
```

File: Week_1_C/cash/cash.c [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    while (change < 0); 
+
+    int convertCents = round(change * 100); 
+    int coins = 0; 
+
+    while (convertCents >= 25) 
+    {
+        convertCents = convertCents - 25;
+        coins++;
+    }
```
Comment (DevBastiS) [inline Week_1_C/cash/cash.c:21]:
```suggestion
    }
    
```

File: Week_1_C/cash/cash.c [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    while (convertCents >= 10) 
+    {
+        convertCents = convertCents - 10;
+        coins++;
+    }
```
Comment (DevBastiS) [inline Week_1_C/cash/cash.c:27]:
```suggestion
    }
    
```

File: Week_1_C/cash/cash.c [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    while (convertCents >= 5) 
+    {
+        convertCents = convertCents - 5;
+        coins++;
+    }
```
Comment (DevBastiS) [inline Week_1_C/cash/cash.c:33]:
```suggestion
    }
    
```

File: Week_1_C/credit/credit.c [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    {
+        if (doubleDigit[m] > 9)
+        {
+            multSum = multSum + ((floor(doubleDigit[m] / 10 )) + (doubleDigit[m] % 10));
+        }
+        else 
+        {        
+            multSum = multSum + doubleDigit[m];
+        }
+    }
```
Comment (DevBastiS) [inline Week_1_C/credit/credit.c:42]:
```suggestion
    }
    
```

File: Week_1_C/credit/credit.c [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+    int nonMultSum = 0;    
+    for (int k = cardLength - 1; k >= 0; k -= 2)
+    {
+        int nonDoubleDigit = cardDigits[k];
+        nonMultSum = nonMultSum + nonDoubleDigit;
+    }
```
Comment (DevBastiS) [inline Week_1_C/credit/credit.c]:
```suggestion
    }
    
```

File: Week_1_C/hello.c [hunk 9]
```diff
@@ -0,0 +1,9 @@
+#include <cs50.h>
+#include <stdio.h>
+
+int main(void)
+{
+    string name = get_string("What is your name: "); //declare a string with 'name' variable
+
```
Comment (DevBastiS) [inline Week_1_C/hello.c]:
```suggestion
```

File: Week_1_C/mario-less.c [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+            printf("#");
+        }
+        printf("  ");
+        for (int right_hash = 0; right_hash < row + 1; right_hash++ )
+        {
+            printf("#");
+        }
+        printf("\n");
+    }
+}
```
Comment (DevBastiS) [inline Week_1_C/mario-less.c]:
```suggestion
}

```

File: Week_1_C/mario-more.c [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+int get_positive_int(string prompt) 
+{
+    int number; //making new integer to copmly with task conditions
+    do // do loop for runing code at least once
+    {
+        number = get_int("%s", prompt);
+    }
+    while (number < 1 || number > 8); // to display digits from 1 to 8 inclusive 
+    return number; //will ask again if anything but 1-8 digits
+}
```
Comment (DevBastiS) [inline Week_1_C/mario-more.c]:
```suggestion
}

```

File: Week_1_C/mario-more.c [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+        }
+        for (int row = 0; row <= height; row++) //loop for rows
+        {
+            printf("#");
+        }
+        printf("\n");
+    }
+}
+
+//get positive number from a user and executing the function
```
Comment (DevBastiS) [inline Week_1_C/mario-more.c]: Remove comments.

**Review by DevBastiS — CHANGES_REQUESTED** [review]

File: Week_1_C/credit/credit.c [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+            case 5:
+                if (cardDigits[1] == 1 || cardDigits[1] == 2 || cardDigits[1] == 3 || cardDigits[1] == 4 || cardDigits[1] == 5)
+                {
+                    printf("MASTERCARD\n");
+                    return 0;
+                }
+                break;
+        }
+    }
+
```
Comment (DevBastiS) [inline Week_1_C/credit/credit.c]:
```suggestion
```

File: Week_1_C/credit/credit.c [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+                    return 0;
+                }
+                break;
+        }
+    }
+    
+           
+    printf("INVALID\n");
+        
+
```
Comment (DevBastiS) [inline Week_1_C/credit/credit.c]:
```suggestion
```

File: Week_1_C/mario-more.c [hunk 15]
```diff
@@ -0,0 +1,35 @@
+#include <cs50.h>
+#include <stdio.h>
+
+int get_positive_int(string prompt);
+
+int main(void)
+
```
Comment (DevBastiS) [inline Week_1_C/mario-more.c]:
```suggestion
```

File: Week_1_C/mario-more.c [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+
+int get_positive_int(string prompt) 
+{
+    int number;
+    do
+    {
+        number = get_int("%s", prompt);
+    }
+    while (number < 1 || number > 8);
```
Comment (DevBastiS) [inline Week_1_C/mario-more.c]:
```suggestion
    while (number < 1 || number > 8); 
    
```

File: Week_1_C/mario-less.c [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+
```
Comment (DevBastiS) [inline Week_1_C/mario-less.c]:
```suggestion

```

**Review by DevBastiS — CHANGES_REQUESTED** [review]

File: Week_1_C/credit/credit.c [hunk 18]
```diff
@@ -0,0 +1,84 @@
...
+                }
+                break;
+
+            case 4:
+                printf("VISA\n");
+                return 0;
+                break;
+         
+            case 5:
+                if (cardDigits[1] == 1 || cardDigits[1] == 2 || cardDigits[1] == 3 || cardDigits[1] == 4 || cardDigits[1] == 5)
```
Comment (stefanluth) [inline Week_1_C/credit/credit.c]: Extract this if-condition to a variable.

**Review by stefanluth — CHANGES_REQUESTED** [review]
Please delete irrelevant files: `.DS_Store` `.vscode/settings.json` `Week_1_C/.DS_Store` `Week_1_C/.vscode/tasks.json`

And move `hello.c` and the `mario` files to separate folders.

File: Week_1_C/cash/cash.c [hunk 19]
```diff
@@ -0,0 +1,43 @@
...
+
+    int convertCents = round(change * 100); 
+    int coins = 0; 
+
+    while (convertCents >= 25) 
+    {
+        convertCents = convertCents - 25;
+        coins++;
+    }
+
```
Comment (DevBastiS) [inline Week_1_C/cash/cash.c:22]:
```suggestion
```

File: Week_1_C/credit/credit.c (hunk 18 above)
Comment (DevBastiS) [reply Week_1_C/credit/credit.c]: Yes please do

File: Week_1_C/mario-less.c (hunk 17 above)
Comment (DevBastiS) [inline Week_1_C/mario-less.c]:
```suggestion

```

File: Week_1_C/mario-more.c (hunk 15 above)
Comment (DevBastiS) [reply Week_1_C/mario-more.c]: Still open

File: Week_1_C/mario-more.c (hunk 16 above)
Comment (DevBastiS) [reply Week_1_C/mario-more.c]: Still open

**Review by DevBastiS — CHANGES_REQUESTED** [review]

File: Week_1_C/credit/credit.c [hunk 20]
```diff
@@ -0,0 +1,81 @@
...
+
+    int nonMultSum = 0;    
+    for (int k = cardLength - 1; k >= 0; k -= 2)
+    {
+        int nonDoubleDigit = cardDigits[k];
+        nonMultSum = nonMultSum + nonDoubleDigit;
+    }
+
+    if ((multSum + nonMultSum) % 10 == 0)
+    {
```
Comment (DevBastiS) [inline Week_1_C/credit/credit.c]: Use "Early-Exit" here.

File: Week_1_C/credit/credit.c [hunk 21]
```diff
@@ -0,0 +1,81 @@
...
+        int nonDoubleDigit = cardDigits[k];
+        nonMultSum = nonMultSum + nonDoubleDigit;
+    }
+
+    if ((multSum + nonMultSum) % 10 == 0)
+    {        
+        switch(cardDigits[0])
+        {
+            case 3:
+                if(cardDigits[1] == 4 || cardDigits[1] == 7)
```
Comment (DevBastiS) [inline Week_1_C/credit/credit.c]: Move this condition to a speaking variable.

File: Week_1_C/mario-more/mario-more.c [hunk 22]
```diff
@@ -0,0 +1,35 @@
...
+
+
+int get_positive_int(string prompt) 
+{
+    int number;
+    do
+    {
+        number = get_int("%s", prompt);
+    }
+    while (number < 1 || number > 8);
```
Comment (DevBastiS) [inline Week_1_C/mario-more/mario-more.c]:
```suggestion
    while (number < 1 || number > 8); 
    
```

**Review by DevBastiS — CHANGES_REQUESTED** [review]

File: Week_1_C/mario-more/mario-more.c (hunk 22 above)
Comment (DevBastiS) [reply Week_1_C/mario-more/mario-more.c]: This is still open I think at least

**Review by DevBastiS — APPROVED** [review]

**Pull request merged**